learn/models/conformal-regression/index.qmd

```diff
@@ -0,0 +1,441 @@
+
+There are a variety of ways to apply this concept (which is unsurprisingly more complex than that the above description). The probably package has implemented a few of them for regression models (classification methods are coming in the future). 
+
+Let's make a simple example to illustrate the results. We'll simulate a data set with a single predictor along with some unknown samples: 
+
+```{r}
+#| label: example-data
+#| out-width: 70%
+#| fig-width: 6
+#| fig-height: 4
+make_data <- function(n, std_dev = 1 / 5) {
+  tibble(x = runif(n, min = -1)) %>%
+    mutate(
+      y = (x^3) + 2 * exp(-6 * (x - 0.3)^2),
+      y = y + rnorm(n, sd = std_dev)
+    )
+}
+
+n <- 1000
+set.seed(8383)
+train_data <- make_data(n)
+
+train_data %>% 
+  ggplot(aes(x, y )) + 
+  geom_point(alpha = 1 / 10)
+```
+
+We'll use these data as a training set and fit a model: 
+
+
+```{r}
+#| label: model-fit
+#| out-width: 70%
+#| fig-width: 6
+#| fig-height: 4
+
+set.seed(484)
+nnet_wflow <- 
+  workflow(y ~ x, mlp(hidden_units = 4) %>% set_mode("regression"))
+
+nnet_fit <- nnet_wflow %>% fit(train_data)
+nnet_pred <- augment(nnet_fit, train_data)
+
+train_data %>% 
+  ggplot(aes(x)) + 
+  geom_point(aes(y = y), alpha = 1 / 10) +
+  geom_line(data = nnet_pred, aes(y = .pred),
+            linewidth = 1, col = "blue")
+```
+
+Let's examine three methods^[A fourth method is implemented for "full conformal prediction." It was developed mainly as a proof of concept. While it is effective, it is _very_ computationally expensive.] to produce prediction intervals: 
+
+## Split Conformal Inference
+
+The most straightforward approach is reserving some data for estimating the residual distribution. We know that simply re-predicting the training set is **a bad idea**; the residuals would be smaller than they should be since the same data are used to create and evaluate the model. 
+
+Let's simulate another data set containing 250 samples and call that the "calibration set". These data can be predicted and the corresponding residuals can be used to define what conforms to model. We'll also create a large test set to see if we've done a good job. 
+
+```{r}
+#| label: more-data
+
+set.seed(7292)
+cal_data  <- make_data(250)
+test_data <- make_data(10000)
+```
+
+The probably package has a set of functions with the prefix `int_conformal` that can be used to create prediction intervals. One is: 
+
+```{r}
+#| label: split-obj
+
+split_int <- int_conformal_split(nnet_fit, cal_data)
+split_int
+```
+
+To get predictions on new data, we use the standard `predict()` method on this object: 
+
+
+```{r}
+#| label: split-pred
+
+# Produce 90% prediction intervals
+test_split_res <- 
+  predict(split_int, test_data, level = 0.90) %>% 
```

Comment:
This smells like it should have a `augment()` method

Reply:
We don't have one implemented yet but that's a great idea.